openqaoa/problems/converters.py

```diff
@@ -0,0 +1,384 @@
+#   Copyright 2022 Entropica Labs
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import numpy as np
+from collections import defaultdict
+from .problem import QUBO
+from ..optimizers.result import Result
+
+
+class FromDocplex2IsingModel:
+    def __init__(self, model, multipliers: float = None):
+
+        """
+        Creates an instance to translate Docplex models to its Ising Model representation
+
+        Parameters
+        ----------
+        model : Docplex model
+            It is an object that has the mathematical expressions (Cost function,
+            Inequality and Equality constraints) of an optimization problem in
+            binary representation.
+
+        """
+        # assign the docplex Model
+        self.model = model.copy()
+
+        # save the index in a dict
+        self.idx_terms = {}
+        for x in self.model.iter_variables():
+            self.idx_terms[x] = x.index
+            if x.vartype.short_name != "binary":
+                TypeError(f"Variable {x.vartype.short_name} is not allowed.")
+        # get doclex qubo and ising model
+        self.qubo_docplex, self.ising_model = self.get_models(multipliers)
+
+    def linear_expr(self, expr):
+        """
+        Adds linear expresions to the objective function stored in self.qubo_dict.
+
+        Parameters
+        ----------
+        expr : model.objective_expr
+            A docplex model attribute.
+
+        """
+        for x, weight in expr.get_linear_part().iter_terms():
+            self.qubo_dict[(self.idx_terms[x],)] += weight
+
+    def quadratic_expr(self, expr):
+        """
+        Adds quadratic terms to the objective function stored in self.qubo_dict.
+
+        Parameters
+        ----------
+        expr : model.objective_expr
+            A docplex model attribute.
+        """
+
+        # save the terms and coeffs of the quadratic part,
+        # considering two index i and j
+        for x, y, weight in expr.iter_quad_triplets():
+            i = self.idx_terms[x]
+            j = self.idx_terms[y]
+            if i == j:  # if this is the term is  Z1**2 for example
+                self.qubo_dict[(i,)] += weight
+            else:
+                self.qubo_dict[(i, j)] += weight
+
+    def Equality2Penalty(self, expression, multiplier: float):
```

Comment:
Why did we choose the camelcase for a class method here?

Reply:
Also, perhaps this method should be a `staticmethod`? I don't see any updates to `self`

Reply:
I will modify the camelcase and the static method, thanks Vishal! :)

Reply:
great, thanks!